### PR TITLE
Updating lint tox rule

### PR DESCRIPTION
**Pull request by dhermes**
Two chief changes

- Moving `run_pylint` and `pylintrc_default` into the `scripts/` directory (along with all other scripts used by `tox`)
- Swapping `pep8_on_repo.sh` for `pep8_on_repo.py` to avoid using `bash`. (We require a script  --Python or bash -- since we only run `pep8` on files that are checked in to the repo by using `git ls-files`. Prior to this change, spurious Python files would cause `tox -e lint` to fail.)

@tseaver I decided to do this partially because of fix-ups for #714 and partially for #1123.

### CODE REVIEW

File: scripts/run_pylint.py [hunk 1]
```diff
@@ -36,8 +36,9 @@
     'docs/conf.py',
     'setup.py',
 ]
-PRODUCTION_RC = 'pylintrc_default'
-TEST_RC = 'pylintrc_reduced'
+SCRIPTS_DIR = os.path.abspath(os.path.dirname(__file__))
+PRODUCTION_RC = os.path.join(SCRIPTS_DIR, 'pylintrc_default')
+TEST_RC = os.path.join(SCRIPTS_DIR, 'pylintrc_reduced')
```
Comment (tseaver) [inline scripts/run_pylint.py]: I don't see the move of `pylintrc_reduced` into the `scripts` subdirectory, unlike the move above of `pylintrc_default`.

File: .gitignore [hunk 2]
```diff
@@ -48,7 +48,7 @@ coverage.xml
 system_tests/local_test_setup
 
 # Make sure a generated file isn't accidentally committed.
-pylintrc_reduced
+scripts/pylintrc_reduced
```
Comment (tseaver) [inline .gitignore]: Remind me:  when does this file get generated?

File: scripts/run_pylint.py (hunk 1 above)
Comment (dhermes) [inline scripts/run_pylint.py]: `pylintrc_reduced` is a generated file, not checked in (see `.gitignore`)

File: .gitignore (hunk 2 above)
Comment (dhermes) [inline .gitignore]: When `run_pylint` runs

**Comment by tseaver** [comment]
LGTM

**Pull request merged**